src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -283,19 +286,17 @@ void reconnect() throws InterruptedException {
 
                 connectError.set(""); // reset on each loop
 
-                if (server.equals(lastServer)) {
-                    this.reconnectWaiter = new CompletableFuture<>();
-                    waitForReconnectTimeout();
-                }
-
                 if (isDisconnectingOrClosed() || this.isClosing()) {
                     break;
                 }
 
+                if (server.equals(lastServer)) {
+                    shouldWait = true;

```

Comment:
if we manage to connect, should shouldWait be set to false?
line 305

Reply:
i think so, otherwise it think we wait if we happen to connect to the last server

Reply:
Yep, good catch, thanks.  I'll fix it, and move the delay up per other comments.